### PR TITLE
MRG: update to code for forthcoming sourmash release

### PR DESCRIPTION
**Pull request by ctb**
This PR is a standing PR to update to the forthcoming sourmash release, probably 0.16.0, which will hopefully include:

- https://github.com/sourmash-bio/sourmash/pull/3344 (not projected to cause any problems ;)
- and (hopefully) https://github.com/sourmash-bio/sourmash/pull/3342 (will need changes here ;))

Fixes https://github.com/sourmash-bio/sourmash_plugin_branchwater/issues/468

TODO:
- [ ] check to see if https://github.com/sourmash-bio/sourmash_plugin_branchwater/issues/398 is fixed by https://github.com/sourmash-bio/sourmash/pull/3342 and this PR. (It probably won't be, but we could use a test anyway!)
- [x] check on fastgather/fastmultigather performance impacts of https://github.com/sourmash-bio/sourmash/pull/3342
- [x] revalidate gather output - virtually guaranteed not to change, but, y'know, better safe than sorry!

### CODE REVIEW

**Comment by ctb** [comment]
gather output validated, as of https://github.com/sourmash-bio/sourmash_plugin_branchwater/commit/2563b0b605af33ddb7f453b4b594c6c9807c2fdf in https://github.com/sourmash-bio/sourmash_plugin_branchwater/pull/430

File: src/utils.rs [hunk 1]
```diff
@@ -478,7 +478,10 @@ pub fn load_sketches_above_threshold(
             if let Ok(against_sig) = against_collection.sig_from_record(against_record) {
                 if let Some(against_mh) = against_sig.minhash() {
                     // downsample against_mh, but keep original md5sum
-                    let against_mh_ds = against_mh.downsample_scaled(query.scaled()).unwrap();
+                    let against_mh_ds = against_mh
+                        .clone()
```
Comment (luizirber) [inline src/utils.rs:482]: would an `.into_minhash()` in `core` that consumes the signature help here? would avoid the need for `.clone()`, which can be substantial for big sketches

Comment (ctb) [reply src/utils.rs:482]: sure!

Comment (luizirber) [reply src/utils.rs:482]: I'll make a PR against https://github.com/sourmash-bio/sourmash/pull/3342

Comment (luizirber) [reply src/utils.rs:482]: PR:https://github.com/sourmash-bio/sourmash/pull/3348

Comment (ctb) [reply src/utils.rs]: thanks! here's what I did; eliminated two (!!) clones,

https://github.com/sourmash-bio/sourmash_plugin_branchwater/pull/467/commits/9b448c8a873e9fde1bc4cb84f441863078677eb9

Comment (ctb) [reply src/utils.rs]: (comments obv welcome ;))

**Comment by ctb** [comment]
Ready for review! Please let me manage merging tho 😭 .

**Review by luizirber — APPROVED** [review]

**Pull request merged**